dvc/command/experiments.py

```diff
@@ -399,8 +407,16 @@ def show_experiments(
         include_params=include_params,
         exclude_params=exclude_params,
     )
-    metric_headers = _normalize_headers(metric_names)
-    param_headers = _normalize_headers(param_names)
+
+    names = {**metric_names, **param_names}
+    counter = Counter(
```

Comment:
for #5989 
The duplicated column name `foo` in the test stage will cause a wrong output in test.

Reply:
@karajan1001, let's do the same for other headers that we have (like `Experiments`, `rev`, etc.) to reduce chances of collision. You can extract the headers from the following to a global variable and reuse them here:

https://github.com/iterative/dvc/blob/b3680b1a9357b206359b70c2753ec6cf9c4e1536/dvc/command/experiments.py#L341-L349

Reply:
Also, should we rename the column `typ` to `Type`?

Reply:
Also, the lowercase names, should we uppercase it?

Reply:
> Also, the lowercase names, should we capitalize it?

for the revs and parent I think we should capitalize them, but for the user-defined ones, capitalization might cause confusion to the users, and make them hard to manage in code ( It is easy to capitalize a string but hard to recover it )

Reply:
@karajan1001, I was only talking about that particular list: {typ, rev, parent}.

Reply:
@skshetry 
But one problem here, "rev" is consisitent with the some other functions for example in `dvc/repo/plots/template.py`, `./dvc/scm/git/__init__.py` and `./dvc/api.py`
while `typ` and `parent` are only used here. So I will only modify  `typ` and `parent` here.

Reply:
But they are not part of a UI, they are mostly part of an API or a schema where that makes sense. Please check `dvc metrics show --all-commits` for example.